src/builders.rs

```diff
@@ -1451,6 +1213,149 @@ impl From<DateBased> for Output {
     /// Create an output logger which defers to the given date-based logger. Use
     /// configuration methods on [DateBased] to set line separator and filename.
     fn from(config: DateBased) -> Self {
-        Output(OutputInner::DateBased { config })
+        let config = log_impl::DateBasedConfig::new(
+            config.line_sep,
+            config.file_prefix,
+            config.file_suffix,
+            if config.utc_time {
+                log_impl::ConfiguredTimezone::Utc
+            } else {
+                log_impl::ConfiguredTimezone::Local
+            },
+        );
+        let computed_suffix = config.compute_current_suffix();
+        // ignore errors - we'll just retry later.
+        let initial_file = config.open_current_log_file(&computed_suffix).ok();
+
+        Output(OutputInner::Logger(Box::new(log_impl::DateBased {
+            config,
+            state: Mutex::new(DateBasedState::new(computed_suffix, initial_file)),
+        })))
+    }
+}
+
+/// A generic formatter that easily provides good defaults while being configurable
+pub struct FormatterBuilder {
+    #[cfg(feature = "colored")]
+    color_config: Option<ColoredLevelConfig>,
+    #[cfg(feature = "chrono")]
+    chrono: bool,
+    level: bool,
+    target: bool,
+}
+
+impl Default for FormatterBuilder {
+    fn default() -> Self {
+        FormatterBuilder {
+            #[cfg(feature = "colored")]
+            color_config: Some(Default::default()),
+            #[cfg(feature = "chrono")]
+            chrono: true,
+            level: true,
+            target: true,
+        }
+    }
+}
+
+impl FormatterBuilder {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    pub fn level(mut self, level: bool) -> Self {
+        self.level = level;
+        self
+    }
+
+    pub fn target(mut self, target: bool) -> Self {
+        self.target = target;
+        self
+    }
+
+    #[cfg(feature = "colored")]
+    pub fn color(mut self, color: bool) -> Self {
+        self.color_config = if color {
+            self.color_config.or_else(Default::default)
+        } else {
+            None
+        };
+        self
+    }
+
+    #[cfg(feature = "colored")]
+    pub fn color_config(
+        mut self,
+        modify_config: impl FnOnce(ColoredLevelConfig) -> ColoredLevelConfig,
+    ) -> Self {
+        self.color_config = self.color_config.map(modify_config);
+        self
+    }
+
+    #[cfg(feature = "chrono")]
+    pub fn chrono(mut self, chrono: bool) -> Self {
+        self.chrono = chrono;
+        self
+    }
+
+    #[rustfmt::skip]
+    pub fn build(
+        self,
+    ) -> impl Fn(FormatCallback<'_>, &std::fmt::Arguments<'_>, &log::Record<'_>) + Sync + Send + 'static
+    {
+        move |out, message, record| {
+            let time = {
+                #[cfg(feature = "chrono")]
+                {
+                    self.chrono
+                        .then(|| chrono::Local::now().format("%Y-%m-%d %H:%M:%S,%3f"))
```

Comment:
Since this could easily become a standard-ish formatter, what would you think of matching the [env_logger](https://docs.rs/env_logger/) format, or at least, matching it as closely as we can given the enable features?

For apps which don't need custom configuration, `env_logger` seems to be fairly standard, and so at least from my perspective, that format (something like `[2017-11-09T02:12:24Z ERROR main]`) is "normal" for rust apps. It didn't used to be my personal favorite, and env_logger was not as popular when `fern` started, so this was not the format I included in the original docs. However, given that it's already fairly normalized now, I think it would make for a pretty safe choice?